packages/backend/src/models/SpaceModel.ts

```diff
@@ -1075,54 +1094,35 @@ export class SpaceModel {
         return this.getFullSpace(spaceUuid);
     }
 
-    async addSpaceAccess(spaceUuid: string, userUuid: string): Promise<void> {
-        const [space] = await this.database('spaces')
-            .select('space_id')
-            .where('space_uuid', spaceUuid);
-
-        const [user] = await this.database('users')
-            .select('user_id')
-            .where('user_uuid', userUuid);
-
-        await this.database(SpaceShareTableName)
+    async addSpaceAccess(
+        spaceUuid: string,
+        userUuid: string,
+        spaceRole: string,
+    ): Promise<void> {
+        await this.database(SpaceUserAccessTableName)
             .insert({
-                space_id: space.space_id,
-                user_id: user.user_id,
+                space_uuid: spaceUuid,
+                user_uuid: userUuid,
+                space_role: spaceRole,
             })
-            .onConflict(['user_id', 'space_id'])
+            .onConflict(['user_uuid', 'space_uuid'])
             .merge();
     }
 
     async removeSpaceAccess(
         spaceUuid: string,
         userUuid: string,
     ): Promise<void> {
-        const [space] = await this.database('spaces')
-            .select('space_id')
-            .where('space_uuid', spaceUuid);
-
-        const [user] = await this.database('users')
-            .select('user_id')
-            .where('user_uuid', userUuid);
-
-        await this.database(SpaceShareTableName)
-            .where('space_id', space.space_id)
-            .andWhere('user_id', user.user_id)
+        await this.database(SpaceUserAccessTableName)
+            .where('space_uuid', spaceUuid)
+            .andWhere('user_uuid', userUuid)
             .delete();
     }
 
     async clearSpaceAccess(spaceUuid: string, userUuid: string): Promise<void> {
```

Comment:
We can remove this method as we don't clear the table when switching between space private/non private.